R/obsPlot.R

```diff
@@ -1,274 +1,484 @@
-                     pointLayer = TRUE,
-                     meanLayer = TRUE,
-                     xlab = groupCol,
-                     ylab = valueCol,
-                     title,
-                     boxColor = "grey30",
-                     boxFill = "deepskyblue3",
-                     boxAlpha = 0.5,
-                     boxNotch = FALSE,
-                     boxNotchWidth = 0.8,
-                     violinColor = "grey30",
-                     violinFill = "goldenrod1",
-                     violinAlpha = 0.5,
-                     pointColor = "grey30",
-                     pointFill = "dodgerblue4",
-                     pointShape = 21,
-                     pointAlpha = 1,
-                     pointSize = 2,
-                     pointJitter = 0,
-                     meanColor = "red2",
-                     meanFill = "goldenrod1",
-                     meanShape = 22,
-                     meanAlpha = 0.7,
-                     meanSize = 3,
-                     baseFontSize = 12,
-                     themeStyle = "grey",
-                     facet = TRUE,
-                     facetRow,
-                     xAngle = 30,
-                     scales = "free_y")
-{
-
-    .addGeoms <- function(MyPlot) {
-        if (boxLayer == TRUE) {
-            MyPlot <- MyPlot + geom_boxplot(alpha = boxAlpha,
-                                            color = boxColor,
-                                            fill = boxFill,
-                                            notch = boxNotch,
-                                            notchwidth = boxNotchWidth,
-                                            outlier.shape = outlier.shape,
-                                            outlier.size = outlier.size)
+obsPlot <- function(DGEdata,
+                    plotType    = "canvasXpress",
+                    designTable = "design",
+                    group       = "replicategroup",
+                    boxLayer    = TRUE,
+                    violinLayer = FALSE,
+                    showPoints  = TRUE,
+                    showMean    = TRUE,
+                    xlab,
+                    ylab,
+                    title,
+                    boxColor    = "deepskyblue3",
+                    boxTransparency = 0.5,
+                    violinColor = "goldenrod1",
+                    violinTransparency = 0.5,
+                    facet       = TRUE,
+                    facetRow,
+                    labelAngle  = 30,
+                    axisFree    = TRUE) {
+    assertthat::assert_that(!missing(DGEdata),
+                            !is.null(DGEdata),
+                            "DGEobj" %in% class(DGEdata),
+                            msg = "DGEdata must be specified and should be of class DGEobj")
+    assertthat::assert_that("counts" %in% names(DGEdata),
+                            msg = "counts matrix must be available in DGEdata to plot the data.")
+    assertthat::assert_that(!is.null(plotType),
```

Comment:
set the default plot to **canvasXpress** after a warning as we discussed with Connie